### PR TITLE
Remove GitHub updater and don't trow exception in release

### PR DESCRIPTION
**Pull request by Bond-009**
This can later be re-implemented if needed, but IMO the application shouldn't handle updates, at most it should show a new version is available. Updating should be handled by the package manager or a separate script.

### CODE REVIEW

**Comment by nvllsvm** [comment]
For Linux users, I agree on updating via third party.

For Windows and macOS users, this functionality would be important to keep. 

Is it possible to keep this behind a feature flag instead?

**Comment by Bond-009** [comment]
macOS has homebrew for updates. As for Windows user they will have to wait untill we implement an alternative to manual updates.
This PR just removes the existing `GitHubUpdater` which doesn't work ATM, and should be replaced with a proper solution.

**Comment by nvllsvm** [comment]
Homebrew is third-party and, while popular, I wouldn't want to count on it being there. Installing dmgs is super convent and would be the go to for a lot of people who don't necessarily understand terminals.

Anyway, I'm good on nuking this for now.

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -2292,48 +2289,12 @@ public void RemovePlugin(IPlugin plugin)
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <param name="progress">The progress.</param>
         /// <returns>Task{CheckForUpdateResult}.</returns>
-        public async Task<CheckForUpdateResult> CheckForApplicationUpdate(CancellationToken cancellationToken, IProgress<double> progress)
+        public Task<CheckForUpdateResult> CheckForApplicationUpdate(CancellationToken cancellationToken, IProgress<double> progress)
```
Comment (nvllsvm) [inline Emby.Server.Implementations/ApplicationHost.cs]: Nuke this instead of stubbing.

**Comment by dkanada** [comment]
Agreed, a feature flag would be ideal. It could be disabled for Linux builds in official repos.

File: Emby.Server.Implementations/ApplicationHost.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/ApplicationHost.cs]: It is needed by the Interface it implements, it doesn't do any harm and makes it easier to re-implement.

Comment (nvllsvm) [reply Emby.Server.Implementations/ApplicationHost.cs]: It's useless code until then and may not be the way we reimplement it.

Can the interface be changed instead?

File: Emby.Server.Implementations/ScheduledTasks/PluginUpdateTask.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using MediaBrowser.Common;
+﻿/*using MediaBrowser.Common;
```
Comment (nvllsvm) [inline Emby.Server.Implementations/ScheduledTasks/PluginUpdateTask.cs]: Remove this file.

File: Emby.Server.Implementations/ScheduledTasks/SystemUpdateTask.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿using MediaBrowser.Common;
+﻿/*using MediaBrowser.Common;
```
Comment (nvllsvm) [inline Emby.Server.Implementations/ScheduledTasks/SystemUpdateTask.cs]: Remove this file.

**Comment by nvllsvm** [comment]
I'm particularly picky about commenting code out "just-in-case" it is reimplemented in the future. More often than not, I've seen code like that sit in production code bases for years without being touched.

**Comment by nvllsvm** [comment]
Right - we will reimplement.

For now, please delete it instead of commenting it out.

**Comment by Bond-009** [comment]
Done

**Pull request merged**